packages/web3torrent/src/App.tsx

```diff
@@ -33,20 +36,45 @@ const App: React.FC = () => {
       }
     });
   }, [initialized]);
+  const [selectedAddress, setSelectedAddress] = useState(undefined);
+  useEffect(() => {
+    if (window.ethereum) {
+      setSelectedAddress(window.ethereum.selectedAddress);
+    }
+  }, []);
 
   useEffect(() => {
     if (window.ethereum && typeof window.ethereum.on === 'function') {
-      const listener = chainId => setCurrentNetwork(Number(chainId));
-      window.ethereum.on('networkChanged', listener);
+      const networkChangeListener = chainId => setCurrentNetwork(Number(chainId));
+      const addressChangeListener = accounts => {
+        console.log(accounts[0]);
```

Comment:
Should we use the pino logger?

Reply:
Ah this should be removed, I just added it for some quick debugging.